### PR TITLE
bug-fix: better configuration of plugin loggers + trace all algod requests

### PR DESCRIPTION
**Pull request by tzaffi**
## Summary

In preparing the bugfix in https://github.com/algorand/go-algorand/pull/5449, I discovered that conduit's logger wasn't printing out at the log-level that I was expecting (TRACE). I made some minor fixes to address this problem and also added the traces that I found helpful. Finally, a couple of logs were missing the received error from algod, so I added these.

## Test Plan

CI + local testing

### CODE REVIEW

File: conduit/pipeline/pipeline.go [hunk 1]
```diff
@@ -102,27 +102,30 @@ func (p *pipelineImpl) registerPluginMetricsCallbacks() {
 	}
 }
 
+// makeConfig creates a plugin config from a name and config pair.
+// It also creates a logger for the plugin and configures it using the pipeline's log settings.
 func (p *pipelineImpl) makeConfig(cfg data.NameConfigPair, pluginType plugins.PluginType) (*log.Logger, plugins.PluginConfig, error) {
 	configs, err := yaml.Marshal(cfg.Config)
 	if err != nil {
-		return nil, plugins.PluginConfig{}, fmt.Errorf("makeConfig(): could not serialize config: %w", err)
+		return nil, plugins.PluginConfig{}, fmt.Errorf("makeLoggerAndConfig(): could not serialize config: %w", err)
```
Comment (tzaffi) [inline conduit/pipeline/pipeline.go:110]:
```suggestion
		return nil, plugins.PluginConfig{}, fmt.Errorf("makeConfig(): could not serialize config: %w", err)
```

File: conduit/pipeline/pipeline.go [hunk 2]
```diff
@@ -102,27 +102,30 @@ func (p *pipelineImpl) registerPluginMetricsCallbacks() {
...
+	lgr.SetFormatter(makePluginLogFormatter(string(pluginType), cfg.Name))
 
 	var config plugins.PluginConfig
 	config.Config = string(configs)
 	if p.cfg != nil && p.cfg.ConduitArgs != nil {
 		config.DataDir = path.Join(p.cfg.ConduitArgs.ConduitDataDir, fmt.Sprintf("%s_%s", pluginType, cfg.Name))
 		err = os.MkdirAll(config.DataDir, os.ModePerm)
 		if err != nil {
-			return nil, plugins.PluginConfig{}, fmt.Errorf("makeConfig: unable to create plugin data directory: %w", err)
+			return nil, plugins.PluginConfig{}, fmt.Errorf("makeLoggerAndConfig: unable to create plugin data directory: %w", err)
```
Comment (tzaffi) [inline conduit/pipeline/pipeline.go:124]:
```suggestion
			return nil, plugins.PluginConfig{}, fmt.Errorf("makeConfig: unable to create plugin data directory: %w", err)
```

File: conduit/pipeline/pipeline.go [hunk 3]
```diff
@@ -102,15 +102,18 @@ func (p *pipelineImpl) registerPluginMetricsCallbacks() {
...
 		return nil, plugins.PluginConfig{}, fmt.Errorf("makeConfig(): could not serialize config: %w", err)
 	}
 
-	l := log.New()
-	l.SetOutput(p.logger.Out)
-	l.SetFormatter(makePluginLogFormatter(plugins.Processor, cfg.Name))
+	lgr := log.New()
+	lgr.SetOutput(p.logger.Out)
+	lgr.SetLevel(p.logger.Level)
+	lgr.SetFormatter(makePluginLogFormatter(string(pluginType), cfg.Name))
```
Comment (tzaffi) [inline conduit/pipeline/pipeline.go:116]: Previously, all plugin loggers were assigned "processor" as their source.

File: conduit/pipeline/pipeline.go [hunk 4]
```diff
@@ -102,15 +102,18 @@ func (p *pipelineImpl) registerPluginMetricsCallbacks() {
...
 	if err != nil {
 		return nil, plugins.PluginConfig{}, fmt.Errorf("makeConfig(): could not serialize config: %w", err)
 	}
 
-	l := log.New()
-	l.SetOutput(p.logger.Out)
-	l.SetFormatter(makePluginLogFormatter(plugins.Processor, cfg.Name))
+	lgr := log.New()
+	lgr.SetOutput(p.logger.Out)
+	lgr.SetLevel(p.logger.Level)
```
Comment (tzaffi) [inline conduit/pipeline/pipeline.go:115]: Inheriting the log level from the pipeline

**Review by shiqizng — APPROVED** [review]

**Review by winder — APPROVED** [review]

**Pull request merged**